### DISCUSSION TITLE
Change fonts in Hugo Scroll theme

### DISCUSSION THREAD

**Discussion by Daniel-The-Fox** [janraasch/hugo-scroll#193] [Q&A]
Hi there, 👋🏻 

I really like the Hugo Scroll theme, many thanks for sharing it! 😄
I noticed that three different fonts are being used throughout the theme.

What's the best approach in order to replace the theme's 3 default fonts by 2 fonts (1 for normal text, 1 for buttons & headlines)?
How can I avoid the `public/fonts` folder from containing fonts not used on the site?

Many thanks in advance! 👍🏻 😃

**Comment by zjedi** [comment]
You would need to delete those fonts from your branch. Other than that, you could leverage assets processing, which should only copy files that are actually used - but this may get a bit tricky.

Yes, fonts need some cleanup, there were some experiments and now we have unused leftovers. There's a ticket #71 for a font supporting wider set of characters, but no suitable font has been found yet.

I think a good way to approach fonts would to use CSS variables like we use for customizing theme colors. Ideally users should be able to swap fonts without having to hard-fork the theme.

**Reply by Daniel-The-Fox** [reply]
Thanks for the answer. 👍🏻

So if I understood you correctly, there’s no ready/easy solution for that currently and I’ll have to manually change the theme’s fonts.css file and manually delete the font files used by the theme so far? 😬🤔

I’m new to Hugo so I don’t know anything about its asset processing. Could you please provide me with a URL to point me in the correct direction? 🙈

**Reply by zjedi** [reply]
Yes, right now easiest way to get rid of extra fonts and/or replacing them with your custom fonts is to do it in your fork. Good that you brought the topic up, this is not really very nice from customization perspective and should be improved. Unfortunately I have no capacity do dig in right now.

Starting point for Hugo Assets is probably Pipes documentation https://gohugo.io/hugo-pipes/introduction/
Assets are only copied if used "programatically" by Hugo, you have to "touch" the resource

**Reply by zjedi** [reply]
This seems to be material to study regarding making the font switchable https://discourse.gohugo.io/t/font-selection-in-theme/8803/4
There is already way to customize CSS, so you can add new fonts by overriding it, but removing the existing definistions would require more radical changes
